File: src/main/java/de/fabilucius/advancedperks/core/SettingsConfiguration.java
package de.fabilucius.advancedperks.core;

import de.fabilucius.advancedperks.core.configuration.Configuration;
import de.fabilucius.advancedperks.core.configuration.annotation.FilePathInJar;

@FilePathInJar("settings.yml")
public class SettingsConfiguration extends Configuration {

    public int getGlobalMaxActivePerks() {
        return this.getInt("global.max_active_perks", -1);
    }

    public boolean isGuiClickSoundsEnabled() {
        return this.getBoolean("gui.click_sounds", true);
    }

    public boolean isMetricsCollectionEnabled() {
        return this.getBoolean("collecting_metrics", true);
    }
}

File: src/main/kotlin/de/fabilucius/advancedperks/core/configuration/type/SettingsConfiguration.kt
package de.fabilucius.advancedperks.core.configuration.type

import com.google.inject.Inject
import com.google.inject.Singleton
import com.google.inject.name.Named
import de.fabilucius.advancedperks.core.configuration.AbstractConfiguration
import de.fabilucius.advancedperks.core.configuration.annotation.FilePathInJar
import de.fabilucius.advancedperks.core.logging.APLogger
import java.io.File

@Singleton
@FilePathInJar("settings.yml")
class SettingsConfiguration @Inject constructor(@Named("configurationDirectory") configDir: File, logger: APLogger) :
    AbstractConfiguration(configDir, logger) {

    init {
        APLogger.debugMode = isPluginInDebugMode()
    }

    fun getGlobalPerkLimit(): Int = getInt("global.max_active_perks")

    fun areGuiClickSoundsEnabled(): Boolean = getBoolean("gui.click_sounds")

    fun shouldMetricsBeCollected(): Boolean = getBoolean("collecting_metrics")

    fun isPluginInDebugMode(): Boolean = getBoolean("debug_mode")

}